contracts/core/OTCWrapperV2.sol

```diff
@@ -0,0 +1,1046 @@
+                _order.isPut
+            );
+        }
+
+        return oToken;
+    }
+
+    /**
+     * @notice settles the vault after expiry
+     * @dev can only be called by the market maker who is the order seller
+     * @param _orderID order id
+     */
+    function settleVault(uint256 _orderID) external nonReentrant {
+        require(orderStatus[_orderID] == OrderStatus.Succeeded, "OTCWrapper: inexistent or unsuccessful order");
+
+        Order memory order = orders[_orderID];
+
+        require(order.seller == _msgSender(), "OTCWrapper: sender is not the order seller");
+
+        UtilsWrapperInterface.ActionArgs[] memory actions = new UtilsWrapperInterface.ActionArgs[](1);
+
+        actions[0] = UtilsWrapperInterface.ActionArgs(
+            UtilsWrapperInterface.ActionType.SettleVault,
+            address(this), // owner
+            order.seller, // address to transfer to
+            address(0), // not used
+            order.vaultID, // vaultId
+            0, // not used
+            0, // not used
+            "" // not used
+        );
+
+        // execute actions
+        controller.operate(actions);
+
+        emit VaultSettled(_orderID);
+    }
+
+    /**
+     * @notice allows user to redeem after expiry
+     * @dev can only be called by the user who is the order buyer
+     * @param _orderID order id
+     */
+    function redeem(uint256 _orderID) external nonReentrant {
+        require(orderStatus[_orderID] == OrderStatus.Succeeded, "OTCWrapper: inexistent or unsuccessful order");
+
+        Order memory order = orders[_orderID];
+
+        require(order.buyer == _msgSender(), "OTCWrapper: sender is not the order buyer");
+
+        // check if vault has enough collateral to cover payout
+        (UtilsWrapperInterface.Vault memory vault, uint256 typeVault, ) = controller.getVaultWithDetails(
+            address(this),
+            order.vaultID
+        );
+
+        (, bool isValidVault) = calculator.getExcessCollateral(vault, typeVault);
+
+        // require that vault is valid (has excess collateral) before redeeming
+        // to avoid allowing redeeming undercollateralized vaults
+        require(isValidVault, "OTCWrapper: insuficient collateral to redeem");
+
+        UtilsWrapperInterface.ActionArgs[] memory actions = new UtilsWrapperInterface.ActionArgs[](1);
+
+        actions[0] = UtilsWrapperInterface.ActionArgs(
+            UtilsWrapperInterface.ActionType.Redeem,
+            address(0), // not used
+            order.buyer, // address to transfer to
+            order.oToken, // otoken address
+            0, // not used
+            order.size, // otoken amount
+            0, // not used
+            "" // not used
+        );
+
+        // execute actions
+        controller.operate(actions);
+
+        orders[_orderID].size = 0;
+
+        emit Reedem(_orderID, order.size);
+    }
+
+    /**
+     * @notice sends fees to beneficiary
+     */
+    function claimFees() external {
```

Comment:
theres a chance in the future that fees are denominated in a non-USDC asset. lets add `_asset` param in claimFees and claim that one. 

can also remove the require as I think safeTransfer takes care of that error

Reply:
https://github.com/ribbon-finance/GammaProtocol-OTC/pull/10/commits/bcbf34e8c2810633eb75d7a1c5ada84ecb0936d5